osu.Game/Modes/Mods/Mod.cs

```diff
@@ -152,10 +146,23 @@ public class ModAutoplay : Mod
         public override double ScoreMultiplier => 0;
         public override Type[] IncompatibleMods => new[] { typeof(ModRelax), typeof(ModSuddenDeath), typeof(ModNoFail) };
 
-        public override void PlayerLoading(Player player)
+        /// <summary>
+        /// Attaches a replay score to a HitRenderer.
+        /// </summary>
+        /// <param name="hitRenderer">The HitRenderer to attach the score to.</param>
+        /// <param name="score">The score to attach.</param>
+        protected void Attach(HitRenderer hitRenderer, Score score)

```

Comment:
this shouldn't be in the `Mod` class. If anything you should make a `ModAutoplay` subclass which handles *all* autoplay attaching. It would only need to expose a `protected Score ReplayScore { get; }`